cmd/dev/ci/deps/url.go

```diff
@@ -0,0 +1,131 @@
+package deps
+
+import (
+	"bytes"
+	"fmt"
+	"github.com/spf13/cobra"
+	"gopkg.in/yaml.v2"
+	"io/ioutil"
+	"log"
+	"os"
+	"text/template"
+)
+
+const ExampleConfigFile = `version: v1.20.2
+url: https://storage.googleapis.com/kubernetes-release/release/{{.Version}}/bin/{{.Os}}/{{.Architecture}}/kubectl
+architecture-mapping:
+  amd64: x64
+os-mapping:
+  darwin: mac
+  linux: unix
+`
+
+type FileNotFoundError struct {
+	Path string
+	Err  error
+}
+
+func (e FileNotFoundError) Error() string { return fmt.Sprintf("Error! Config file not found: '%s'", e.Path) }
+func (e FileNotFoundError) Unwrap() error { return e.Err }
+
+type InvalidFileError struct {
+	Path string
+	Err  error
+}
+
+func (e InvalidFileError) Error() string { return fmt.Sprintf("Error! Config file '%s' is invalid:\n%s\n\nExample of a valid config file:\n%s", e.Path, e.Err.Error(), ExampleConfigFile) }
+func (e InvalidFileError) Unwrap() error { return e.Err }
+
+type Component struct {
+	Version string `yaml:"version"`
+	Url string `yaml:"url"`
+	ArchitectureMapping ArchitectureMapping `yaml:"architecture-mapping"`
+	OsMapping OsMapping `yaml:"os-mapping"`
+	Os string `yaml:"os,omitempty"`
+	Architecture string `yaml:"architecture,omitempty"`
+}
+type ArchitectureMapping struct {
+	AMD64 string `yaml:"amd64"`
+	ready bool
+}
+
+type OsMapping struct {
+	Darwin string `yaml:"darwin"`
+	Linux string `yaml:"linux"`
+	ready bool
+}
+
+func (component Component) String() string {
+	d, err := yaml.Marshal(component)
+	if err != nil {
+		log.Fatalf("error: %v", err)
+	}
+	return fmt.Sprintf("%s", string(d))
+}
+
+func (component *Component) getComponent(data []byte) *Component{
```

Comment:
In go, method receivers are usually singular characters. Most often the first character of the struct the method belongs to, so here:

```suggestion
func (c *Component) getComponent(data []byte) *Component{
```

It also appears that this method is unused :)

Reply:
Yeah, I will remove it. I used it only in the beginning for testing. And I will adjust the naming of the method receivers.